modules/terraform-zscc-asg-lambda-aws/main.tf

```diff
@@ -0,0 +1,284 @@
+resource "aws_iam_policy" "lambda_logs_policy" {
+  description = "Policy which permits lambda to create and send logs"
+  name        = "${var.name_prefix}-asg-lambda-send-logs-${var.resource_tag}"
+  policy      = data.aws_iam_policy_document.lambda_logs_policy_document.json
+}
+
+# Attach lambda logs policy to IAM Role
+resource "aws_iam_role_policy_attachment" "lambda_logs_attachment" {
+  policy_arn = aws_iam_policy.lambda_logs_policy.arn
+  role       = aws_iam_role.asg_lambda_iam_role.name
+}
+
+
+################################################################################
+# Create Lambda Function
+################################################################################
+resource "aws_lambda_function" "asg_lambda_function" {
+  function_name = "${var.name_prefix}_asg_lambda_function_${var.resource_tag}"
+  handler       = "${var.name_prefix}_asg_lambda_function_${var.resource_tag}.lambda_handler"
+  runtime       = "python3.9"
+  filename      = "${path.module}/${var.asg_lambda_filename}"
+  role          = aws_iam_role.asg_lambda_iam_role.arn
+  timeout       = 180
+  memory_size   = 256
+
+  environment {
+    variables = {
+      ASG_NAMES              = jsonencode(var.autoscaling_group_names)
+      CC_URL                 = var.cc_vm_prov_url
+      SECRET_NAME            = var.secret_name
+      HC_DATA_POINTS         = "10"
+      HC_UNHEALTHY_THRESHOLD = "7"
+    }
+  }
+
+  tags = merge(var.global_tags)
+}
+
+
+################################################################################
+# Create Cloudwatch Event Rules and Targets for Scheduler
+################################################################################
+resource "aws_cloudwatch_event_rule" "asg_cloudwatch_scheduler_event_rule" {
+  name                = "${var.name_prefix}-cc-asg-scheduled-event-rule-${var.resource_tag}"
+  description         = "EventBridge rule to trigger Lambda function every 10 minutes"
+  schedule_expression = "rate(10 minutes)"
+}
+
+resource "aws_cloudwatch_event_target" "asg_cloudwatch_scheduler_event_target" {
+  rule      = aws_cloudwatch_event_rule.asg_cloudwatch_scheduler_event_rule.name
+  target_id = aws_lambda_function.asg_lambda_function.function_name
+  arn       = aws_lambda_function.asg_lambda_function.arn
+}
+
+
+################################################################################
+# Create Cloudwatch Event Rules and Targets for Lifecycle changes
+################################################################################
+resource "aws_cloudwatch_event_rule" "asg_cloudwatch_lifecycle_event_rule" {
+  name        = "${var.name_prefix}-cc-asg-lifecycle-event-rule-${var.resource_tag}"
+  description = "Event rule for EC2 Instance-terminate Lifecycle Action"
+
+  event_pattern = <<PATTERN
+{
+  "source": ["aws.autoscaling"],
+  "detail-type": ["EC2 Instance-terminate Lifecycle Action"],
+  "detail": {
+    "AutoScalingGroupName": ${jsonencode(var.autoscaling_group_names)}
```

Comment:
are you sure this will take multiple autoscaling group names and the rule will trigger based on this value? Or should it be one rule per ASG created?